### PR TITLE
Update releasing-islandora.md

### PR DESCRIPTION
**Pull request by kstapelfeldt**
Adding modifications as per the September 29th tech call to add information about Semantic versioning. Other small edits (such as changing Danny's organization).

## Purpose / why

> _**Replace this text** - Restate the purpose/justification of this work and include links to any Issues or other discussions that are related to this work._

## What changes were made?

> _**Replace this text** - State clearly the direct additions or modifications made in this pull request._

## Verification

> _**Replace this text** - Document the details that help a reviewer verify the documentation._

## Interested Parties

> _**Replace this text** - Name some folks who may be interested, if documentation related mention @Islandora/documentation, or, if unsure, @Islandora/8-x-committers_

---

## Checklist

> __Pull-request reviewer__ should ensure the following

* [ ] Does this PR link to related [issues](https://github.com/Islandora/documentation/issues/)?
* [ ] Does the proposed documentation align with the [Islandora Documentation Style Guide](https://islandora.github.io/documentation/contributing/docs_style_guide/)?
* [ ] Are the changes accurate, useful, free of typos, etc?
* [ ] Does this PR update the _last updated on_ date on the documentation page?

> __Person merging__ should ensure the following
* [ ] Does mkdocs still build successfully? (This is indicated by TravisCI passing. To test locally, and see warnings, see [How To Build Documentation](https://islandora.github.io/documentation/technical-documentation/docs-build/).)
* [ ] If pages are renamed or removed, have all internal links to those pages been fixed?
* [ ] If pages are added, have they been linked to or placed in the menu?
* [ ] Did the PR receive at least one approval from a committer, and all issues raised have been addressed?

### CODE REVIEW

File: docs/contributing/releasing-islandora.md
Comment (manez) [inline docs/contributing/releasing-islandora.md]: Super minor, but I think it's actually "Born-Digital"

File: docs/contributing/releasing-islandora.md
Comment (seth-shaw-unlv) [inline docs/contributing/releasing-islandora.md]: We need a space between "Drupal's approach" and "and support".

Use "element" instead of "section" (language taken from semver.org).

File: docs/contributing/releasing-islandora.md
Comment (seth-shaw-unlv) [inline docs/contributing/releasing-islandora.md]: I would rename this to something like "Undoing a Release" drop this section to the bottom. This should rarely occur, if ever.

File: docs/contributing/releasing-islandora.md
Comment (seth-shaw-unlv) [inline docs/contributing/releasing-islandora.md]: Spelling: "rleeases" -> "releases"

File: docs/contributing/releasing-islandora.md
Comment (seth-shaw-unlv) [inline docs/contributing/releasing-islandora.md]: "Version numbers can also be incremented and a new release authored." -> "Instead, increment the version number and tag a new release."

**Review by seth-shaw-unlv — CHANGES_REQUESTED** [review]
Most of these items are relatively small changes.

However, I'm also recommending a larger change: I think we should drop all the subsections under #releasing-php-code. This section was written with the assumption that we are doing a "full release" where *everything* is getting new version numbers. It should be revised based on the idea that each of the code repositories can be released on their own terms.

I would also include a new section (either on this page or elsewhere) where we talk about managing composer files. Specifically, code repo maintainers need to be aware of this note I put on the Semantic Versioning proposal:

> **Note to module maintainers specifying composer requirements:** please familiarize yourself with the [Next Significant Release Modifiers](https://getcomposer.org/doc/articles/versions.md#next-significant-release-operators) so that module dependencies are as permissive as possible. Most often dependency versions should be tagged using the caret (`^`) so that bug-fixes and minor releases can be pulled in without updating their module’s composer file.


This whole release-components-as-needed process only works well if we manage our composer dependencies well.

**Comment by seth-shaw-unlv** [comment]
I just noticed that *we already have* a page on versioning that needs to be updated: https://islandora.github.io/documentation/technical-documentation/versioning/

**Comment by seth-shaw-unlv** [comment]
So, parts of the new semantic versioning narrative should be added to the versioning page and the release page mostly needs to call attention to that versioning policy page. We can then add notes about maintaining the `composer.json` file on the versioning page.

File: docs/contributing/releasing-islandora.md
Comment (rosiel) [inline docs/contributing/releasing-islandora.md:114]: Missing this header!

Comment (kstapelfeldt) [reply docs/contributing/releasing-islandora.md:114]: I think @seth-shaw-unlv  wanted it removed?

Comment (seth-shaw-unlv) [reply docs/contributing/releasing-islandora.md:114]: I don't recall making that request.

Comment (rosiel) [reply docs/contributing/releasing-islandora.md:114]: It was a larger re-org piece about releasing things separately vs all repos at once. It stands. If we push this through in an acceptable state, i can look at tweaking it to better communicate what is implied by individual releases.

**Comment by rosiel** [comment]
> I just noticed that _we already have_ a page on versioning that needs to be updated: https://islandora.github.io/documentation/technical-documentation/versioning/

see above PR #1912 

> However, I'm also recommending a larger change: I think we should drop all the subsections under #releasing-php-code. This section was written with the assumption that we are doing a "full release" where everything is getting new version numbers. It should be revised based on the idea that each of the code repositories can be released on their own terms.

This is true. I can take on that rewrite so it's not holding up @kstapelfeldt's improvements. I'm also working on a [message to the community](https://docs.google.com/document/d/1E7jf-MuKo_z4GLKxC25idS9qB8nlCo3DmZ2LZGAxw5o/edit) about how this SemVer stuff means things will be released independently; don't be alarmed when our minor version numbers increase with Firefox-like speed... Feedback welcome; I'll pass it by the Tech Call before sending it out.

**Comment by rosiel** [comment]
@seth-shaw-unlv would you feel okay if we merge this improvement, then draft an even better version with the Tech Call's help?

**Comment by seth-shaw-unlv** [comment]
Sorry for the delay. We were hit with emergency server issues that I was dealing with yesterday and today.

Yes, we can probably merge this and then do further editing later, although there some branch conflicts that @kstapelfeldt need to resolve before I can.

**Review by seth-shaw-unlv — APPROVED** [review]

**Pull request merged**
